R/summary.pop_data.R

```diff
@@ -12,44 +13,130 @@
 #' library(dplyr)
 #'
 #' xs_data <- load_pop_data("https://osf.io/download//n6cp3/")
+#' summary(xs_data, strata = "Country")
 #'
-#' summary(xs_data)
-#'
-summary.pop_data <- function(object, ...) {
-  ages <-
-    object %>%
-    distinct(.data$id, .data$age)
-
-  cat("\nn =", nrow(ages), "\n")
-
-  cat("\nDistribution of age: \n\n")
-  age_summary <-
-    ages %>%
-    pull("age") %>%
-    summary() %>%
-    print()
-
-  cat("\nDistributions of antigen-isotype measurements:\n\n")
-
-  ab_summary <-
-    object %>%
-    dplyr::summarize(
-      .by = .data$antigen_iso,
-      Min = object %>% get_value() %>% min(na.rm = TRUE),
-      `1st Qu.` = object %>% get_value() %>% quantile(.25, na.rm = TRUE),
-      Median = object %>% get_value() %>% median(),
-      `3rd Qu.` = object %>% get_value() %>% quantile(.75, na.rm = TRUE),
-      Max = object %>% get_value() %>% max(na.rm = TRUE),
-      `# NAs` = object %>% get_value() %>% is.na() %>% sum()
-    ) %>%
-    as.data.frame() %>%
-    print()
-
-  to_return <- list(
-    n = nrow(ages),
-    age_summary = age_summary,
-    ab_summary = ab_summary
-  )
-
-  return(invisible(to_return))
+summary.pop_data <- function(object, strata = NULL, ...) {
+  # get relevant columns from object
+  age_column <- object %>% get_age_var()
+  value_column <- object %>% get_value_var()
+  id_column <- object %>% get_id_var()
+
+  if (is.null(strata)) {
+    ages <-
+      object %>%
+      distinct(
+        .data[[age_column]],
+        .data[[id_column]]
+      )
+
+    age_summary <-
+      ages %>%
+      pull(age_column) %>%
+      summary() %>%
+      print()
+
+    cat("\nn =", nrow(ages), "\n")
+
+    cat("\nDistribution of age: \n\n")
```

Comment:
why did the order of these commands change? what does the output look like? Please provide before-and-after screenshots.

Reply:
The commands need not change. I have switched them back to the original place.